### ISSUE TITLE
Translate content to Japanese

### ISSUE THREAD

**Issue by bazzadp** [HTTPArchive/almanac.httparchive.org#586]
These are the chapters to be translated, and they exist in the [content directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content):

**2020**
- [ ] capabilities

**2019**
- [x] accessibility - @MSakamaki 
- [x] caching - @ksakae1216
- [x] cdn - @ksakae1216
- [x] cms - @ksakae1216 (requires review)
- [x] compression - @ksakae1216 (requires review)
- [x] css - @ksakae1216 (requires review)
- [x] ecommerce - @ksakae1216 (requires review)
- [x] fonts - @ksakae1216  (requires review)
- [x] http2 - @ksakae1216 - #661
- [x] javascript - @ksakae1216 (requires review) 
- [x] markup - @MSakamaki 
- [x] media - @ksakae1216 (requires review)
- [x] mobile-web - @ksakae1216 (requires review)
- [x] page-weight- @ksakae1216 (requires review)
- [x] performance - @MSakamaki 
- [x] pwa - @ksakae1216 (requires review)
- [x] resource-hints - @ksakae1216 (requires review)
- [x] security - @ksakae1216 (requires review)
- [x] seo - @MSakamaki 
- [x] third-parties - @ksakae1216 (requires review)

Additionally the following pages need translated too in the [language specific templates directory](https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/templates):
- [x] base.html - @ksakae1216 (requires review)
- [x] base_chapter.html - @ksakae1216 (requires review)
- [x] contributors.html - @ksakae1216 (requires review)
- [x] error.html
- [x] index.html - @ksakae1216 (requires review)
- [x] featured_chapters.html  - @ksakae1216 (requires review)
- [x] methodology.html - @ksakae1216 (requires review)
- [x] table_of_contents.html - @ksakae1216 (requires review)
- [x] accessibility_statement.html - @ksakae1216 (requires review)

There is no need to translate the chapters HTML pages are they are generated off the markdown combined with the above templates.

Common notes for writing consistency are here: https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Translators'-Guide. Feel free to edit that and/or add Japanese-specific extras by editing this comment.

Japanese specific extra advice:

* Nayori of foreign words
https://www.jtca.org/standardization/katakana_guide_3_20171222.pdf
https://www.bunka.go.jp/kokugo_nihongo/sisaku/joho/joho/kijun/naikaku/gairai/index.html
* English, katakana, hiragana, and kanji should be carefully selected throughout the content.
* Use these [textlint](https://github.com/textlint/textlint) settings: https://gist.github.com/MSakamaki/a3b7cbf57231b6287a19be28534cb50d

**Comment by bazzadp** [comment]
@ksakae1216 has expressed an interest in helping out.

@ksakae1216, for a start it would be great if you could review @MSakamaki 's work so far in https://github.com/HTTPArchive/almanac.httparchive.org/tree/master/src/templates/ja/2019/chapters and see if you spot any issues with the translations?

After that I've sent you an invite to the team and when you accept that, you should be able edit the first comment in this issue and put your name beside a chapter,

To do the translations, clone the repo, grab the markdown from the /en/ folder and copy it to the /ja/ folder and edit it, then submit a pull request with the translation and @MSakamaki can review for language and I (or someone else) will also review it from a non-language perspective and then we'll merge it into master.

Let me know if there are any questions!

**Comment by ksakae1216** [comment]
@bazzadp, Thank you.

I have a question.
What should I review first?
I looked at the link but didn't know what to review.

**Comment by bazzadp** [comment]
@ksakae1216 , ideally we'd have a second person reviewing all translations before we merge them to master. It's amazing what a second pair of eyes can spot. So anything you translate, @MSakamaki will review, and similarly anything he translates you can review. And if we get a third (or fourth!) Japanese translator then we can spread the load.

However when @MSakamaki translated those first three chapters (Markup, Performance and SEO), we didn't have a second Japanese translator so we just merged them without that review. We were hoping someone would join before we launched them so we could do that review - and now you're here! 😀

So it would be great if you could read those three translations now, and if you spot any issues then open a PR with suggested fixes. @MSakamaki can then review and he agrees approve the PR and we'll merge it in. If you're happy with the translations as they are and see no need for any changes then just add a comment to let us know.

One thing the French translation team did, that I really liked, was come up with a set of conventions for the French team to use. I copied those to the bottom of the [first comment](https://github.com/HTTPArchive/almanac.httparchive.org/issues/586#issue-533812318), but they are things like don't translation certain terms, which gender to use, and to use French versions of MDN documents whether they exist... etc. I don't know if those ones are exactly relevant to Japanese so feel free to discard them, but I'm sure there are similar conventions you and @MSakamaki should agree on. So as you read @MSakamaki 's translations I think it would be good to come up with some of these Japanese conventions too, so there's some consistency to the translations no matter who translated it. And actually documenting them in this issue will ensure these conventions are clear and help if another translator joins the team in future, or if someone queries any translations after go live.

Once you've read the three already translated chapters and have a sense of how to translate chapters, feel free to grab your own chapter by putting your name beside an unassigned chapter in the [first comment](https://github.com/HTTPArchive/almanac.httparchive.org/issues/586#issue-533812318) and away you go!

Let me know if that all makes sense? However, these are all just suggestions. I've never done any translations and speak no Japanese, so you and @MSakamaki are the ones who can drive this issue through to completion - if you have better ways of working then go for it! Some of the translations teams are using one GitLocalise repo for all the translations, some have chat rooms (there's a [Slack Channel for the Almanac](https://httparchive.slack.com/archives/CGFLV2S06) - let me know if you can't access and need an invite), or you can message each other on Twitter or any other messaging platform, or just use this issue to comment back and forth. Up to you two!

**Comment by bazzadp** [comment]
Oh just realised I gave the wrong link! That's proibably why you're confused. Try this one: https://github.com/HTTPArchive/almanac.httparchive.org/tree/master/src/content/ja/2019 (and edited above comment).

**Comment by ksakae1216** [comment]
@bazzadp , Thank you proposal.

I check three translated chapters.

Next, I will discuss to @MSakamaki whether rules should be created like the French team.

**Comment by MSakamaki** [comment]
Hello @ksakae1216 !

As for the translation rules for Japanese, I am currently working on the following rules:

Nayori of foreign words
https://www.jtca.org/standardization/katakana_guide_3_20171222.pdf
https://www.bunka.go.jp/kokugo_nihongo/sisaku/joho/joho/kijun/naikaku/gairai/index.html

Japanese proofreading (automation recommended, I useing Tool)
https://marketplace.visualstudio.com/items?itemName=ICS.japanese-proofreading

There are other free OSSs, such as textlint and Redpen. If you have something you want to use, please let me know.

The other point is that English, katakana, hiragana, and kanji should be carefully selected throughout the content.

Feel free to talk about any other rules that might be needed.

**Comment by ksakae1216** [comment]
Hi @MSakamaki 

Thanks advice.

I see.

**Comment by MSakamaki** [comment]
@ksakae1216 
After translating accessibility, I tried to change Japanese proofreading to [textlint](https://github.com/textlint/textlint).
Using this, I would like to match the expressions as much as possible.

Your settings can be made as shown in the following link.
https://gist.github.com/MSakamaki/a3b7cbf57231b6287a19be28534cb50d

Do you have any opinions?

**Comment by ksakae1216** [comment]
@MSakamaki 

No problem.

I try add vscode plugin.

**Comment by ksakae1216** [comment]
@bazzadp 

I have a question.

"Figure 4." of the following document has no png file. Is it OK?
https://github.com/HTTPArchive/almanac.httparchive.org/blob/master/src/content/en/2019/seo.md

**Comment by bazzadp** [comment]
> "Figure 4." of the following document has no png file. Is it OK?
> https://github.com/HTTPArchive/almanac.httparchive.org/blob/master/src/content/en/2019/seo.md

This is not really a figure but more highlighting of a particular state (often called a [pull quote](https://en.wikipedia.org/wiki/Pull_quote)):

![image](https://user-images.githubusercontent.com/10931297/71613895-acacf800-2ba0-11ea-9ff4-d46a7003c5c3.png)

We have basically 4 types of "figures":

1. Tables of stats we have gathered, like [Figure 11 in the SEO chapter](https://almanac.httparchive.org/en/2019/seo#fig-11). It could be argued these are "tables" rather than "figures" but we've called them "figures" in the Almanac when they are reporting stats we have gathered.
2. Graphics like [Figure 1 in the SEO chapter](https://almanac.httparchive.org/en/2019/seo#fig-1). Stats. Note these have a PNG and also have `data-` attributes that allow us to create an interactive embed of the Google Sheets graph with a bit of JavaScript.
3. Pull quotes like you have noted in [Figure 4 in the SEO chapter](https://almanac.httparchive.org/en/2019/seo#fig-4). It could be argued these are just quotes rather than figures, but we've decided to call them figures.
4. External images we have imported from other sources. Like [Figure 2 in the Markup chapter](https://almanac.httparchive.org/en/2019/markup#fig2).

Hope that makes sense but let us know if you've any other questions!

**Comment by ksakae1216** [comment]
@bazzadp 
Thanks, Explanation figures.

**Comment by bazzadp** [comment]
@ksakae1216 , @MSakamaki would either of you have some time to look at the templates in src/templates/en/2019 and translate to src/templates/jp/2019?

We're looking to launch the chapters that are already translated in #671 but to do that, and let others benefit from your hard work so far, we need the core templates translated.

I've made this a good bit easier in #673 so the templates should mostly be lots of place holders like this:

```
{% block description %}The Web Almanac is an annual state of the web report combining the expertise of the web community with the data and trends of the HTTP Archive.{% endblock %}
```

You just need to translate all the parts between the `{% block XXX %}` and `{% endblock %}` and then the chapter generation script will automatically use these phrases as appropriate.

**Comment by bazzadp** [comment]
Btw it’s [base.html](https://github.com/HTTPArchive/almanac.httparchive.org/blob/master/src/templates/en/2019/base.html) and [base_chapter.html](https://github.com/HTTPArchive/almanac.httparchive.org/blob/master/src/templates/en/2019/base_chapter.html) that are most important to be able to release the chapters you’ve translated already. These are also quite short so should be easy to translate. The rest (the home page, the table of content pages... etc.) will take longer and can be added at a later time.

**Comment by ksakae1216** [comment]
@bazzadp , Sorry delay response.

I looked base.html and base_chapter.html.
Maybe don't need translate?

Is my understanding correct?

**Comment by bazzadp** [comment]
I cobbled together something and merged it. Would be good if you could review and translate anyone's I couldn't?

**Comment by ksakae1216** [comment]
I intended to ask if I needed to do any of the following:
> @ksakae1216 , @MSakamaki would either of you have some time to look at the templates in src/templates/en/2019 and translate to src/templates/jp/2019?

But you don't have to work because it's done, right?
https://github.com/HTTPArchive/almanac.httparchive.org/pull/690

**Comment by bazzadp** [comment]
Yes but I only partially did them in #690 .

If you look at the contents of those files they are mostly still English. But I did find a few Japanese translations from the other files so not 100% English but definitely not 100% Japanese either 😀So if you want to finish the translations of those files to also translate the phrases I didn't, then that would be good.

But think with #690 we have enough to launch if you don't have time for that right now.

**Comment by ksakae1216** [comment]
I see.
I will translate the untranslated files little by little.

**Comment by bazzadp** [comment]
@ksakae1216 I added you to the [Contributors page](https://almanac.httparchive.org/ja/2019/contributors?teams=translators) in https://github.com/HTTPArchive/almanac.httparchive.org/commit/42c710325f4cf6568ad220b69dd4b0cc13d4bd3f. Hope that’s OK and feel free to submit a PR to correct or add any more details.

**Comment by bazzadp** [comment]
@ksakae1216 do you fancy tackling these three next in the templates directory?

- base.html - partially done to allow us to launch Japanese chapters
- base_chapter.html - partially done to allow us to launch Japanese chapters
- index.html - partially done to allow us to launch Japanese chapters

Think it would look better for our Japanese readers if the home page and all the core text was translated. Can then return to the remaining chapters after that?

**Comment by ksakae1216** [comment]
@bazzadp, Yes, I will translate 3 html file.

**Comment by bazzadp** [comment]
Thanks @ksakae1216 , just pushed the updated index and fonts chapters to live:
https://almanac.httparchive.org/ja/2019/
https://almanac.httparchive.org/ja/2019/fonts

Looks much better now the Japanese home page is actually in Japanese! 😀

**Comment by ksakae1216** [comment]
@bazzadp, @MSakamaki.
 I translate resource-hints.

**Comment by bazzadp** [comment]
> @bazzadp, @MSakamaki.
> I translate resource-hints.

@ksakae1216 please be aware I made a small tweak to the Resource Hints chapter in https://github.com/HTTPArchive/almanac.httparchive.org/pull/731 to move some long code snippets to their own lines. Just merged there now so make sure you do same in your translation.

Barry

**Comment by ksakae1216** [comment]
@bazzadp, @MSakamaki.
I will translate media next.

**Comment by ksakae1216** [comment]
@bazzadp, @MSakamaki .

I will translate contributors.html, featured_chapters.html, methodology.html.

**Comment by ksakae1216** [comment]
@bazzadp.

if @MSakamaki finish to translate javascript, Do we finish?

**Comment by bazzadp** [comment]
Looks like! Excellent work.

However @MSakamaki hasn't been contributing in a while so not sure if he's making any progress on the JavaScript chapter. Maybe you could have a go and can compare if @MSakamaki does come back to the project?

**Comment by ksakae1216** [comment]
@bazzadp, @MSakamaki 

OK, I will translate javascript chapter tomorrow.

**Comment by ksakae1216** [comment]
@bazzadp , We finished translate.
Thank you so much for reviewing it.

**Comment by bazzadp** [comment]
Thank YOU so much for all the hard work! Well done on being the first fully translated language!

The last chapter is now live: https://almanac.httparchive.org/ja/2019/javascript

Ideally we'll get @MSakamaki (or another Japanese speaker) to review your translations at some point but will close this ticket in the meantime. We can reopen if if we need to.

**Comment by bazzadp** [comment]
Reopening for 2020 chapters. Probably should be a separate issue but as Japanese is only language that made the finish line with all the 2019 content, I've added the 2020 chapters to their issues so it's easier to be consistent.

**Comment by ksakae1216** [comment]
I will start translate on sataday or sunday(11/28 or 11/29)

**Comment by ksakae1216** [comment]
@bazzadp  I report to you about capabilities.md.

1. I found not enough 'js' 204, 232 row.
2.  What is "pf" of "pf page loads" 290 row?

**Comment by rviscomi** [comment]
@ksakae1216 thanks for pointing those out! I've fixed those typos in #1591.

**Comment by ksakae1216** [comment]
@rviscomi, @bazzadp 
I have a question.
I found mistake.
318 line en/2020/javascript.md.
```
Once again, this appears to be an area where third-party scripts are actually doing better than first-party scripts. If we break the compression methods out by first- and third-party, we see that 24% of third-party scripts have Brotli applied, compared to only 15% of third-party scripts.
```
I think "15% of **first-party** scripts."

**Comment by bazzadp** [comment]
Agreed. Raised #1930 to fix this and a few other things I spotted. You might want to look at that after it's merged if you're mid-translation.

**Comment by bazzadp** [comment]
Those fixes have now been merged in @ksakae1216

**Comment by ksakae1216** [comment]
@bazzadp I have a question.
I translate 2020/accessibility.md.
Line 136, I think that first and second description is revers.

> The **first** description tells the user what to expect if they activate the control, whereas the **second** just describes its appearance and has an unnecessary file extension, both of which are irrelevant to the meaning of the image.

↓↓↓

> The **second** description tells the user what to expect if they activate the control, whereas the **first** just describes its appearance and has an unnecessary file extension, both of which are irrelevant to the meaning of the image.

**Comment by bazzadp** [comment]
Not sure I agree. Here’s the full quote:

> For example, a red arrow icon button that has the action of moving to a new step in the interface could be described as “continue to step 3 of 5” rather than “red arrow png”. The first description tells the user what to expect if they activate the control, whereas the second just describes its appearance and has an unnecessary file extension, both of which are irrelevant to the meaning of the image.

To make that more verbose, so it’s more obvious what that means to you, it would be:

> So the first description (“continue to step 3 of 5”) tells the user what to expect if they activate the control. The second description (“red arrow png”) just describes its appearance and has an unnecessary file extension. Both the appearance (the fact it’s a red arrow), and the extension (the fact it’s a png), are irrelevant to the meaning of the image - a sighted user knows the arrow means move forward. So the intent of the image (what the image represents) is more important in this case, than what the actual image is.

Does that make sense now?

**Comment by ksakae1216** [comment]
Thanks, I misstake.

**Comment by bazzadp** [comment]
No worries. Keep raising these sorts of things though please!

For one, it’s important you understand for the translation itself.

For second, translators are usually (but not always) right when they raise things like this and help fix errors. I love that we get an extra round of reviews with the translations - it massively improves the quality for English and all future translations! At the very least it implies the wording should be reviewed to avoid any misunderstanding in future (though having done that here, I don’t see any immediate need or way to improve it).

**Comment by ksakae1216** [comment]
@bazzadp Sorry, I have a question.
2020/http2.md
Is line 82 and 102 no problem? 
"**" forget close?
line 82
`<td class="numeric">**34.47%</td>`

line 102
`** As with last year's crawl ....`

**Comment by bazzadp** [comment]
That's actually OK. Here the stars mean "see note below for more explanation on this figure". And then we have the note below also marked with two stars.

So it's not the opening of bold in markdown as I think you think it is.

Pretty cool that our markdown to HTML converter recognising this for what it is and doesn't put the whole thing in bold between these two!

**Comment by ksakae1216** [comment]
@bazzadp Thanks, I see.

**Comment by bazzadp** [comment]
FYI @HTTPArchive/japanese-translators just got the regular "April Search performance for almanac.httparchive.org" from Google and looks like the 2019 Caching chapter in Japanese got a lot of visibility this month.

Not sure why to be honest, but good that people are coming to the site for the translations! And good that 2020 caching looks to be next on the list if there is interest in it!

**Comment by bazzadp** [comment]
Oh and btw, here's the dashboard in case you're ever curious how many readers are benefiting from your hard work:
https://datastudio.google.com/u/0/reporting/1c89f8c3-7e11-4376-8815-39966e13180e/page/1M?params=%7B%22df37%22:%22include%25EE%2580%25800%25EE%2580%2580IN%25EE%2580%2580%252Fja%252F%22%7D

**Comment by ksakae1216** [comment]
Hello @hi-toshi
How progress performance chapter?

Please tell me. 😄

**Comment by tunetheweb** [comment]
Just one chapter to go!!! amazing work @ksakae1216 !!

Also dunno if you spotted it but there is also this one:

- [ ] 2020 / base.html - foreword added later

**Comment by ksakae1216** [comment]
@tunetheweb OK.
I will translate 2020 / base.html.

**Comment by ksakae1216** [comment]
@bazzadp OK, I will translate next week.

**Comment by tunetheweb** [comment]
We added a small note to our Accessibility Statement about colour contrast issues:

https://github.com/HTTPArchive/almanac.httparchive.org/blob/14f85122c88d14ee1f8dd0474479424338766625/src/templates/es/accessibility_statement.html#L69-L71

Would be great if someone could translate this (@ksakae1216 ?). Very small and easy!

**Comment by ksakae1216** [comment]
@tunetheweb OK.
I will translate at Today or Tommorow.

**Comment by tunetheweb** [comment]
@ksakae1216 could you translate this and I'll tweet it out from the HTTPArchive Twitter account?:

> The JavaScript chapter of the 2021 Web Almanac has been translated to Japanese by @ beltway7.
> https://almanac.httparchive.org/ja/2021/javascript
> Follow the RSS feed to get notified of new translations as they are released: https://almanac.httparchive.org/ja/rss.xml

**Comment by ksakae1216** [comment]
@tunetheweb I translated it. Thank you.

2021年版Web AlmanacのJavaScriptの章が@beltway7さんによって日本語に翻訳されました。
https://almanac.httparchive.org/ja/2021/javascript
RSSフィードをフォローすると、新しい翻訳がリリースされたときに通知されます: https://almanac.httparchive.org/ja/rss.xml

**Comment by ksakae1216** [comment]
@tunetheweb I have a question.
almanac.httparchive.org/src/content/en/2021/fonts.md not exist, So I don't translate ja/2021/fonts.md right?

**Comment by tunetheweb** [comment]
Correct. Fonts dropped out this year unfortunately. Will update the issues to remove it.

**Comment by tunetheweb** [comment]
Good spot! Too much copying and pasting from the previous years chapter last year 😁

Could you change this link in the 2020 text to this instead?:

```
[last year's](../2020/compression#fig-9)
```

There isn't a heading near it in 2020 so let's link to the figure directly instead.

**Comment by ksakae1216** [comment]
Thanks your response.
I'll fix that.

**Comment by ksakae1216** [comment]
@tunetheweb I want to help for me.

1. I did finish fetch and merge
![Screen Shot 2022-07-18 at 10 55 57](https://user-images.githubusercontent.com/1982567/179435151-b798902c-4102-4a7a-8b54-204e4590287e.png)
 
2. But exist diff. Why exist diff. Do you what causes diff to exist?
![Screen Shot 2022-07-18 at 10 56 13](https://user-images.githubusercontent.com/1982567/179435306-ed174a38-e202-4380-a97b-49c084569f5a.png)

**Comment by tunetheweb** [comment]
You did a fetch and merge. So your `main` must have been ahead, and now you've merged in the latest changes from this repo.

Notice this:

<img width="221" alt="image" src="https://user-images.githubusercontent.com/10931297/179491012-f5b90e25-83ca-4744-814c-1c10ea0ac811.png">

So yes you are not behind - but you're actually ahead now!

Try not to make any changes to your `main` branch. That way you will always just be behind, and can then bring it up to speed again.

To be honest I [force reset my `main` branch](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Git-Guide#commands-to-force-reset-your-main-branch-to-upstream), rather than fetch and merge into it. That way I always know it's ONLY got the changes from the main repo and never any of my own local changes.

In fact I even have a couple of helpful aliases I've created to make this easier:

```
gforcemain='git fetch upstream;git checkout main;git reset --hard upstream/main;'
gforcepushmain='git push origin main --force'
```

So my workflow when I want to work on new work, is:
1. Force reset main (`gforcemain`) - NOTE this will blast away ALL changes on your `main` branch.
2. Push to GitHub if I want to check I've done that right (`gforcepushmain`)
3. Create a new branch from this up to date `main` (`git checkout -b my-new-feature-branch`)
4. Make changes
5. Merge in changes from upstream `main` periodically (`git fetch upstream; git merge upstream/main`)
6. Push changes to GitHub periodically.
7. Open PR
8. Repeat steps 4-6 as many times as I want.

I can reset my `main` branch each time before starting new work - even if I haven't completed the old work on another branch. Having the `main` branch separate without any work, allows me to reset and get a clean branch each time.

**Comment by ksakae1216** [comment]
@tunetheweb Thanks, I understanded well for your careful explanation.

**Comment by tunetheweb** [comment]
Hey @ksakae1216 are you planning on translating the 2021 foreword and the 2021 Methodology (a lot of this is similar to 2020)? If so that would allow us to print a copy of 2021 edition in Japanese and send you a copy.

Also we have some 2022 chapters available if interested?

**Comment by ksakae1216** [comment]
Hi @tunetheweb
OK, I will translate the 2021 foreword, the 2021 Methodology and 2022 chapters

**Comment by tunetheweb** [comment]
Amazing work again @ksakae1216 ! Could you translate the 2024 foreword and then I think we're done!
